File: Leetcode/2138 - Divide a String Into Groups of Size k/solution.js
var divideString = function (s, k, fill) {
    let result = [];
    let currStr = "";
    for (let i = 0; i < s.length; i++) {
        currStr += s[i];
        if (currStr.length === k) {
            result.push(currStr);
            currStr = "";
        }
    }
    if (currStr.length) {
        while (currStr.length !== k) {
            currStr += fill;
        }
        result.push(currStr);
    }
    return result;
};

File: Leetcode/2138 - Divide a String Into Groups of Size k/solution.py
def divideString(self, s: str, k: int, fill: str) -> List[str]:
    result: List[str] = []
    curr_str: str = ""

    for char in s:
        curr_str += char
        if len(curr_str) == k:
            result.append(curr_str)
            curr_str = ""

    if curr_str:
        while len(curr_str) < k:
            curr_str += fill
        result.append(curr_str)

    return result